src/services/replyService.js

```diff
@@ -13,6 +13,34 @@ const {
 }                = require('../utils/constants');
 
 module.exports = {
+  broadcastAndStoreReply: function({ authenticatedUserInstance, postId, body }) {
+    return PostModel.findById(postId)
+      .then(postRecord => {
+        const postAuthor = postRecord.userId;
+        const postPermLink = postRecord.permLink;
+        const permLink = this.replyPermlink(postAuthor, postPermLink, authenticatedUserInstance.username)
+        return [authenticatedUserInstance.broadcastReply({ postAuthor, postPermLink, permLink, body }), permLink];
+      })    
+      .spread((broadcastSuccess, permLink) => {
+        if (broadcastSuccess) {
+          return [authenticatedUserInstance.userInOurDb, permLink];
+        }
+      })
+      .spread((userRecord, permLink) => {
+        return this.fetchSingleSteemitReply(postId, permLink, userRecord);
+      })
+      .catch(err => {
+        new Error(err.error_description);
+      })
+  },
+
+  // $ curl -s --data '{"jsonrpc":"2.0", "method":"condenser_api.get_content_replies", "params":["trevonjb", "because-i-do-care"], "id":1}' https://api.steemit.com | jq 
+  replyPermlink: function(postAuthor, postPermLink, commenter) {
+     // TODO: add created timestamp to reply permLink
+     // "oadissin-re-trevonjb-because-i-do-care-20180911t021129058z"
+     return [commenter, 're', postAuthor, postPermLink].join('-')
```

Comment:
Minor suggestion. It's generally good practice to put operations like this in variables because it makes the code easier to read, and with good naming, it's much easier for the developer to read and know what's going on. It would be good to put the `array.toString` in a descriptive variable and return the variable.

Reply:
ok